app/src/main/java/io/appium/uiautomator2/core/AccessibilityNodeInfoHelpers.java

```diff
@@ -96,16 +108,18 @@ public static Rect getVisibleBounds(@Nullable AccessibilityNodeInfo node) {
         Rect screen = new Rect(0, 0, uiDevice.getDisplayWidth(), uiDevice.getDisplayHeight());
         ret.intersect(screen);
 
-        // Find the visible bounds of our first scrollable ancestor
-        for (AccessibilityNodeInfo ancestor = node.getParent(); ancestor != null; ancestor = ancestor.getParent()) {
+		// Find the visible bounds of our first scrollable ancestor 
+		for (AccessibilityNodeInfo ancestor = node.getParent(); ancestor != null && depth < MAX_DEPTH; ancestor = ancestor.getParent()) {
             // If this ancestor is scrollable
             if (ancestor.isScrollable()) {
                 // Trim any portion of the bounds that are hidden by the non-visible portion of our
                 // ancestor
-                Rect ancestorRect = getVisibleBounds(ancestor);
+                Rect ancestorRect = getVisibleBounds(ancestor, depth);
                 ret.intersect(ancestorRect);
                 break;
             }
+			
+			depth++;
```

Comment:
depth should be incremented before the condition

Reply:
Done